atom/browser/web_contents_preferences.cc

```diff
@@ -134,15 +137,24 @@ WebContentsPreferences::~WebContentsPreferences() {
 
 bool WebContentsPreferences::SetDefaultBoolIfUndefined(
     const base::StringPiece& key,
-    bool val) {
+    bool val,
+    Status status) {
   auto* current_value =
       preference_.FindKeyOfType(key, base::Value::Type::BOOLEAN);
   if (current_value) {
     return current_value->GetBool();
   } else {
     preference_.SetKey(key, base::Value(val));
-    return val;
+
+    if (status == Status::Deprecated && web_contents_) {
+      auto internal_contents = atom::api::WebContents::CreateFrom(
+          v8::Isolate::GetCurrent(), web_contents_);
+      internal_contents->Emit("-deprecated-default",
```

Comment:
Would be nice if a `version` argument was also emitted here so that the warning message could read "is being removed in version 4.0.0" instead of at some undefined point in the future.

Might help to motivate downstream apps to update their defaults sooner